internal/cache/util/util.go

```diff
@@ -99,3 +101,29 @@ func IsCacheHandleInvalid(readErr error) bool {
 		strings.Contains(readErr.Error(), ErrInSeekingFileHandleMsg) ||
 		strings.Contains(readErr.Error(), ErrInReadingFileHandleMsg)
 }
+
+func CreateCacheDirectoryIfNotPresentAt(dirPath string) error {
+	_, statErr := os.Stat(dirPath)
+
+	if os.IsNotExist(statErr) {
+		err := os.MkdirAll(dirPath, FileDirPerm)
+		if err != nil {
+			return fmt.Errorf("error in creating directory structure %s: %v", dirPath, err)
+		}
+		return nil
```

Comment:
I think we shouldn't return here. Doing that will add additional write permission check which is fine ?

Reply:
yes, I was also in dilemma to have a redundant check or not, I chose this one as I wanted to avoid additional check in order to avoid increasing the mount time, but I am okay with both.